src/layers/SelectionLayer.js

```diff
@@ -22,70 +20,24 @@ import { DrawLineStringHandler } from '@nebula.gl/layers/dist/mode-handlers/draw
 import { Draw90DegreePolygonHandler } from '@nebula.gl/layers/dist/mode-handlers/draw-90degree-polygon-handler';
 import { SplitPolygonHandler } from '@nebula.gl/layers/dist/mode-handlers/split-polygon-handler';
+import { DrawRectangleHandler } from '@nebula.gl/layers/dist/mode-handlers/draw-rectangle-handler';
 import { DrawRectangleUsingThreePointsHandler } from '@nebula.gl/layers/dist/mode-handlers/draw-rectangle-using-three-points-handler';
 import { DrawCircleFromCenterHandler } from '@nebula.gl/layers/dist/mode-handlers/draw-circle-from-center-handler';
 import { DrawCircleByBoundingBoxHandler } from '@nebula.gl/layers/dist/mode-handlers/draw-circle-by-bounding-box-handler';
 import { DrawEllipseByBoundingBoxHandler } from '@nebula.gl/layers/dist/mode-handlers/draw-ellipse-by-bounding-box-handler';
 import { DrawEllipseUsingThreePointsHandler } from '@nebula.gl/layers/dist/mode-handlers/draw-ellipse-using-three-points-handler';
 
-class DrawRectangleByDraggingHandler extends ModeHandler {
-  constructor() {
-    super();
-    this.corner1 = undefined;
-    this.isDragging = false;
-  }
-
-  handleStartDragging(event) {
-    const result = { editAction: null, cancelMapPan: false };
-    this.isDragging = true;
-    const corner1 = event.groundCoords;
-    const corner2 = event.groundCoords;
-    this._setTentativeFeature(bboxPolygon([corner1[0], corner1[1], corner2[0], corner2[1]]));
-    this.corner1 = corner1;
-    return result;
-  }
-
-  handlePointerMove(event) {
-    const result = { editAction: null, cancelMapPan: false };
-    const { isDragging, corner1 } = this;
+import { DrawRectangleMode, DrawPolygonMode, ViewMode } from '@nebula.gl/edit-modes';
 
-    if (!isDragging || !corner1) {
-      // nothing to do yet
-      return result;
-    }
 
-    const corner2 = event.groundCoords;
-    this._setTentativeFeature(bboxPolygon([corner1[0], corner1[1], corner2[0], corner2[1]]));
-    return result;
-  }
-
-  handleStopDraggingOrClick(event) {
-    const result = { editAction: null, cancelMapPan: false };
-    const { isDragging, corner1 } = this;
-
-    if (!isDragging || !corner1) {
-      // nothing to do yet
-      return result;
-    }
-
-    const corner2 = event.groundCoords;
-    this._setTentativeFeature(bboxPolygon([corner1[0], corner1[1], corner2[0], corner2[1]]));
-    const tentativeFeature = this.getTentativeFeature();
-    const editAction = this.getAddFeatureOrBooleanPolygonAction(tentativeFeature.geometry);
-    this._setTentativeFeature(null);
-    this.corner1 = undefined;
-    this.isDragging = false;
-    return editAction;
-  }
-
-  handleStopDragging(event) {
-    return this.handleStopDraggingOrClick(event);
-  }
+const MODE_MAP = {
+  [SELECTION_TYPE.RECTANGLE]: DrawRectangleMode,
+  [SELECTION_TYPE.POLYGON]: DrawPolygonMode,
+};
 
-  handleClick(event) {
-    return this.handleStopDraggingOrClick(event);
-  }
-}
+const MODE_CONFIG_MAP = {
+  [SELECTION_TYPE.RECTANGLE]: { dragToDraw: true },
```

Comment:
but don't we need to click to draw?

Reply:
Yes it would make sense that `drawToDraw: true` would enable dragging to draw, as the docs state https://github.com/uber/nebula.gl/blob/master/docs/api-reference/modes/overview.md#modeconfig-1 but it does not seem to do so. 

And if we want the click-to-draw functionality anyway, we can just remove passing any `modeConfig` at all, which does seem to work.